### ISSUE TITLE
REQUEST - Cease Fire/Resume Fire toggle for Beam Weapons

### ISSUE THREAD

**Issue by Wrongtown** [daid/EmptyEpsilon#205]
I propose a toggle button be added to the **Weapons** station (maybe near the existing **LOCK** button?) with the following values:
**DISABLE BEAMS**
**ENABLE BEAMS**

Beams should default to being on initially (**DISABLE BEAMS** visible) so new players aren't stuck wondering why their beams aren't firing. If they use the toggle to switch them off and can't figure out how to reverse it I'm not sure there's much hope for them.

**Why?**
Currently the Weapons officer doesn't have full control of the beam weapons. For example, they cannot:
1. Cease fire on a vessel (exception: if there are other vessels around they *can* select them instead).
2. Hold fire on a vessel until the right moment.
3. Fire beams on ships or stations that aren't "enemies".

I can see some cool moments emerging (especially in PvP) if weapons has the option to target a ship (or even better, a subsystem!) but hold off on firing until the time is right.

Of course wanting to fire on "Friendly" ships is a bit niche, but I don't really think it's a big problem to give them the option if they have the ability to prevent firing just because they selected a ship. If I'm honest some of the funniest "Captain shouting at a crewmen" moments I've seen have been the result of a stray nuke, so allowing the same mistake with beams doesn't seem so bad to me :wink:

**Comment by Fouindor** [comment]
Great suggestion. Maybe a simpler solution would be that clicking on no target would select no target.

However, I'm not too fond of firing on friendlies, it would cause too much confusion.

**Comment by Fouindor** [comment]
However, as Tactical and Single Pilot also control steering, it may be better only to enable that on Weapons.

**Comment by nallath** [comment]
You could also have some form of system much like the elder scrolls games have; If you walk around with your weapon drawn, some npc's might have reactions to that. 
Flying around some traders with your weapons powered up is pretty suspicious behaviour ;)

**Comment by Wrongtown** [comment]
Nice one @nallath! I think that might be better served with actual Power Detection, raised #208 as a separate issue to explain :smile:

**Comment by nallath** [comment]
I don't think that npc ships actually have a power system at the moment. They just fake it.

**Comment by Wrongtown** [comment]
Fair enough, in the case of AI ships I guess you'd also fake the power on/off flag. Overwhelmingly they would have Power On to all the things whenever players are close right?

**Comment by oznogon** [comment]
Worth noting: On the Weapons station, clicking empty space on the radar breaks the current target lock, which resolves the cease/hold fire aspects of this issue.

Tactical and Single Pilot stations still can't break a target lock, and there's no way for player ships to fire beams at non-hostile ships of unknown faction or friendly ships.
